### ISSUE TITLE
API docs, aliases, synonyms

### ISSUE THREAD

**Issue by drkibitz** [GoodBoyDigital/pixi.js#1565]
What are we going for in terms of the API documentation? Are you planning on documenting the API as only available on the `PIXI` namespace? Or are you going to document the module namespaces. IMO I would prefer everything be documented in their respective module namespaces, then where things are exposed as a convenience on `PIXI` just use a `@see {@link blah}` tag.

Now that I see things as they are, putting everything on the `PIXI` namespace, this is hard to follow and document. I should have voiced my opinion on the original issue, but I would have definitely preferred namespaces now that I'm trying to make more contributions.

**Comment by englercj** [comment]
So everything will be exposed on the PIXI namespace for convenience. As much as I believe the namespacing to be the Right Way™, I think we are in the minority of that mindset. If not, here is the place to prove me wrong community!

Everything is documented [in its namespace][doc] right now since that was how it was exposed, but since we changed the [main export][main] to only export on the root the docs need to now be updated.

[doc]: https://github.com/GoodBoyDigital/pixi.js/blob/dev/src/extras/MovieClip.js#L22
[main]: https://github.com/GoodBoyDigital/pixi.js/blob/dev/src/index.js

**Comment by drkibitz** [comment]
Right now, this is what is available on PIXI object. Seems like this will probably just grow, and grow... :disappointed: 

PIXI.VERSION
PIXI.RENDERER_TYPE
PIXI.BLEND_MODES
PIXI.SCALE_MODES
PIXI.RETINA_PREFIX
PIXI.RESOLUTION
PIXI.FILTER_RESOLUTION
PIXI.DEFAULT_RENDER_OPTIONS
PIXI.SHAPES
PIXI.SPRITE_BATCH_SIZE
PIXI.utils
PIXI.math
PIXI.CONST
PIXI.DisplayObject
PIXI.Container
PIXI.Stage
PIXI.DisplayObjectContainer
PIXI.Sprite
PIXI.ParticleContainer
PIXI.SpriteRenderer
PIXI.ParticleRenderer
PIXI.Graphics
PIXI.GraphicsData
PIXI.GraphicsRenderer
PIXI.Texture
PIXI.BaseTexture
PIXI.RenderTexture
PIXI.VideoBaseTexture
PIXI.CanvasRenderer
PIXI.CanvasGraphics
PIXI.CanvasBuffer
PIXI.WebGLRenderer
PIXI.ShaderManager
PIXI.Shader
PIXI.AbstractFilter
PIXI.autoDetectRenderer
PIXI.PI_2
PIXI.RAD_TO_DEG
PIXI.DEG_TO_RAD
PIXI.Point
PIXI.Matrix
PIXI.Circle
PIXI.Ellipse
PIXI.Polygon
PIXI.Rectangle
PIXI.RoundedRectangle
PIXI.Ticker
PIXI.MovieClip
PIXI.TilingSprite
PIXI.cacheAsBitmap
PIXI.getChildByName
PIXI.Mesh
PIXI.Rope
PIXI.MeshRenderer
PIXI.MeshShader
PIXI.FXAAFilter
PIXI.SpriteMaskFilter
PIXI.AsciiFilter
PIXI.BloomFilter
PIXI.BlurFilter
PIXI.BlurXFilter
PIXI.BlurYFilter
PIXI.ColorMatrixFilter
PIXI.ColorStepFilter
PIXI.ConvolutionFilter
PIXI.CrossHatchFilter
PIXI.DisplacementFilter
PIXI.DotScreenFilter
PIXI.GrayFilter
PIXI.DropShadowFilter
PIXI.InvertFilter
PIXI.NoiseFilter
PIXI.NormalMapFilter
PIXI.PixelateFilter
PIXI.RGBSplitFilter
PIXI.ShockwaveFilter
PIXI.SepiaFilter
PIXI.SmartBlurFilter
PIXI.TiltShiftFilter
PIXI.TiltShiftXFilter
PIXI.TiltShiftYFilter
PIXI.TwistFilter
PIXI.InteractionData
PIXI.InteractionManager
PIXI.interactiveTarget
PIXI.Loader
PIXI.bitmapFontParser
PIXI.spineAtlasParser
PIXI.spritesheetParser
PIXI.textureParser
PIXI.loader
PIXI.Spine
PIXI.SpineRuntime
PIXI.Text
PIXI.BitmapText
PIXI.SpriteBatch
PIXI.AssetLoader

**Comment by photonstorm** [comment]
I'm confused - what was the point of spending all that time and effort breaking it down into modules to just glob it all into a single namespace at the end of it?

This is a v3 release, right? You won't get a chance to modify the API again on this scale until v4.

At the very least break-away the Filters and Parsers?

**Comment by englercj** [comment]
@photonstorm I totally agree, like a lot. I feel like having them in their respective namespace is cleaner, more organized, and promotes discoverability. Putting them in a single namespace just doesn't make sense.

**Comment by GoodBoyDigital** [comment]
yeah the above totally highlights how silly this is getting :)

fear not though! we have a plan..

at the top level we will expose core, maths and consts. 

everything else will be in its namespace.

We will also move text into core as we consider this to be a core element.

This means all the common elemnts will be nice and easy to access. Whilst all the other items that are used less frequently can sit happy in their little namespace homes.

:+1:

**Comment by englercj** [comment]
I have made the changes necessary to put plugins back into their namespaces:

a605fe25305df5a8b1ec6e79ee9a537971db33f8

Thanks all for the discussion!
